discoverd2/server/http.go

```diff
@@ -75,23 +75,21 @@ type httpAPI struct {
 	Store Datastore
 }
 
-func jsonError(w http.ResponseWriter, code int, err error) {
-	httphelper.JSON(w, code, struct {
-		Error string `json:"error"`
-	}{err.Error()})
+func jsonError(w http.ResponseWriter, code hh.ErrorCode, err error) {
+	hh.Error(w, hh.JSONError{Code: code, Message: err.Error()})
```

Comment:
Should we maybe just fold this error message unwrapping behavior into the `hh.Error` function instead of the "Something went wrong" default text?